### PR TITLE
add route prefix option

### PR DESCRIPTION
**Pull request by heinosasshallik**
Also made all indentation 2-space to keep consistent style

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,3 +58,14 @@ server.register({
...
 ```
+
+Path prefix
+----------------------
+You may specify a path prefix for all routes by supplying the `pathPrefix` option.
+
+```
+const routeOptions = {
+  routes: './routes/**/*.js',
+  pathPrefix: '/some-prefix',
```
Comment (robertpallas) [inline README.md]: Just `prefix` is used in Hapi docs, lets keep it simple.

File: index.js [hunk 2]
```diff
@@ -43,6 +43,10 @@ module.exports = {
         route = resolveRouteImport(`${globOptions.cwd}/${file}`);
         route = _.defaultsDeep(route, options.defaultRoute, defaultRoute);
 
+        if (options.pathPrefix) {
+          route.path = options.pathPrefix + route.path;
```
Comment (robertpallas) [inline index.js]: Isn't there a way that route.path does not start with `/`. Just for security would be good to stitch this path together in smarter way with a function that adds slashes if not there.

**Review by robertpallas — CHANGES_REQUESTED** [review]
Small ideas.

**Comment by heinosasshallik** [comment]
I thought it would also be a good idea to add some validation to the route prefix to avoid confusing bugs in case the developer fails to provide the prefix in the correct format. So I added that as well.

File: index.js [hunk 3]
```diff
@@ -43,8 +43,11 @@ module.exports = {
         route = resolveRouteImport(`${globOptions.cwd}/${file}`);
         route = _.defaultsDeep(route, options.defaultRoute, defaultRoute);
 
-        if (options.pathPrefix) {
-          route.path = options.pathPrefix + route.path;
+        if (options.prefix) {
+          if (route.path.charAt(0) !== '/') {
```
Comment (robertpallas) [inline index.js]: I like `path.posix.join` from https://stackoverflow.com/questions/16301503/can-i-use-requirepath-join-to-safely-concatenate-urls

**Review by robertpallas — CHANGES_REQUESTED** [review]
Also add basic tests.

**Review by robertpallas — APPROVED** [review]

**Pull request merged**